From b309af488c6e5ca88216c9fc8f137a8018d493fb Mon Sep 17 00:00:00 2001
From: Shunqian Zheng <zhengsq@rock-chips.com>
Date: Wed, 22 Jul 2020 10:52:15 +0800
Subject: [PATCH] net: wireless: rockchip: rkwifi: bcmdhd: fix gcc8 warning

drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c:6975:39:
warning: argument to 'sizeof' in 'strncpy' call is the same expression
as the source; did you mean to use the size of the destination?
[-Wsizeof-pointer-memaccess]
error, forbidden warning:dhd_linux.c:6975
   strncpy(drvname, info.driver, sizeof(info.driver));

Signed-off-by: Shunqian Zheng <zhengsq@rock-chips.com>
Change-Id: I8f39153a2df4bb2395cd9aac9bd763587f509713
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
index 81da661da2a5..6df687c16bde 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
@@ -6972,7 +6972,7 @@ dhd_ethtool(dhd_info_t *dhd, void *uaddr)
 		/* Copy out any request driver name */
 		if (copy_from_user(&info, uaddr, sizeof(info)))
 			return -EFAULT;
-		strncpy(drvname, info.driver, sizeof(info.driver));
+		strncpy(drvname, info.driver, sizeof(drvname));
 		drvname[sizeof(info.driver)-1] = '\0';
 
 		/* clear struct for return */
-- 
2.35.3

